auditbeat/module/auditd/config.go

```diff
@@ -92,6 +100,11 @@ var defaultConfig = Config{
 	StreamBufferConsumers:  0,
 }
 
+// IsEnabled returns true if the enable field is set to true in the config.
+func (r *Reload) IsEnabled() bool {
+	return r != nil && (r.Enabled == true || r.Period != 0)
```

Comment:
According to this logic, the following config will result in reload being enabled:
```yaml
reload:
  enabled: false
  period: 1m
```

This is misleading. I understand that what you want is for this to be implicitly enabled:
```yaml
reload:
  period: 1m
```

For that you can:
1. Remove the enabled flag and let the value of `period` to control if the feature is enabled or not.
2. Set enabled=true by default in `defaultConfig` above, and change the `||` in this logic to a `&&`.
3. Make Enabled a *bool instead of a bool, then you can:
`return r != nil && r.Period!=0 && (r.Enabled == nil || *r.Enabled == true)`

Reply:
Sorry for the late reply. After having the `defaultConfig`, the logic looks much cleaner for me.